src/main/java/core/basesyntax/service/strategy/strategyimpl/PurchaseOperation.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+
+        if (currentQuantity == null) {
+            throw new IllegalArgumentException("Fruit does not exist in storage.");
+        }
+        if (currentQuantity < 0) {
+            throw new IllegalArgumentException("Invalid quantity for purchase.");
+        }
```

Comment:
you can use `currentQuantity < transaction.getQuantity()` and simplify this method ;)